samples/wifi/provisioning/sample_description.rst

```diff
@@ -0,0 +1,96 @@
+.. _wifi_provisioning_sample_desc:
+
+Sample description
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This sample demnstrates how to provision device with Nordic Semiconductor's Wi-Fi chipsets over BLE.
+
+Requirements
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+The sample also requires an Android or iOS device with nRF Wi-Fi Provisioner
+application installed:
+
+#. `Andriod App <https://play.google.com/store/apps/details?id=no.nordicsemi.android.wifi.provisioning>`_
+#. `iOS App <https://apps.apple.com/us/app/nrf-wi-fi-provisioner/id1638948698>`_
+
+Overview
+********
+
+The goal of Wi-Fi Provisioning Service is to provide a method that allows
+user to provision Wi-Fi device, which typically lacks input/output capability.
+
+The service can be largely divided into three parts:
+
+#. Component that handles provisioning-related tasks and events
+#. Transport layer used to exchange information between the device and the configurator
+#. Component to handle the Provisioning data (in RAM and flash) which  will be accessed
+   by multiple threads
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/wifi/provisioning`
+
+If you want to generate .pb.h and .pb.c files when building the project,
+CONFIG_WIFI_PROVISIONING_PROTO_BUILD_TIME_GENERATION needs to be y. If you
+want to use pre-generated .pb.h and .pb.c files, it needs to be n. This Kconfig
+option is disabled by default.
+
+Testing
+=======
+
+|test_sample|
+
+1. |connect_kit|
+#. |connect_terminal|
+#. After initialization, you can see output like::
+
+       wifi_prov: Advertising successfully started
+
+#. Open nRF Wi-Fi Provisioner application, and enbale Bluetooth if it is disabled.
+#. Search for and connect to the device you want provision. If connection and pairing
+   is done, you can see output like::
+
+       wifi_prov: Connected
+       wifi_prov: Pairing Complete
+
+#. After connection and pairing, you can conduct AP scan, input password if required,
+   and trigger Wi-Fi connection attempt.
+#. If connection succeeds, you can see output like::
+
+       wpa_supp: wlan0: CTRL-EVENT-CONNECTED - Connection
+       to <AP MAC Address> completed [id=0 id_str=]
+
+Dependencies
+============
+
+This sample uses the following `sdk-nrfxlib`_ library:
+
+* :ref:`nrfxlib:nrf_security`
+
+This sample also uses modules that can be found in the following locations
+in the |NCS| folder structure:
+
+* ``modules/lib/nanopb``
```

Comment:
```suggestion
This sample also uses modules that can be found in the following locations
in the |NCS| folder structure:

* ``modules/lib/nanopb``
```

Reply:
We are not using nanopb based generation in this PR, so better not mention it anywhere now.

Reply:
The runtime library is used (encode/decode.h). And CONFIG_NANOPB has to be y. So I think we need to specify here.